File: library/src/main/java/net/anapsil/mvvmbase/test/RepositoryTest.java
package net.anapsil.mvvmbase.test;

import android.util.Log;

import org.junit.After;
import org.junit.Before;

import java.io.IOException;
import java.io.InputStream;

import okhttp3.OkHttpClient;
import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.MockWebServer;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RepositoryTest {
    protected final MockWebServer mockServer = new MockWebServer();
    protected MockResponse mockResponse;

    @Before
    public void setup() throws IOException {
        mockServer.start();
        mockResponse = new MockResponse();
    }

    @After
    public void tearDown() throws IOException {
        mockServer.close();
    }

    protected Retrofit getRetrofit(OkHttpClient okHttpClient) {
        return new Retrofit.Builder()
                .baseUrl(mockServer.url("/").toString())
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build();
    }

    protected String getJson(String path) {
        String json = null;
        try {
            InputStream is = getClass().getClassLoader().getResourceAsStream(path);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException ex) {
            Log.e("getJson", ex.getMessage());
        }

        return json;
    }
}

File: library/src/main/java/net/anapsil/mvvmbase/test/RepositoryTest.kt
package net.anapsil.mvvmbase.test

import android.util.Log
import okhttp3.OkHttpClient
import okhttp3.mockwebserver.MockResponse
import okhttp3.mockwebserver.MockWebServer
import org.junit.After
import org.junit.Before
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import java.io.IOException
import java.nio.charset.Charset

open class RepositoryTest {
    protected val mockServer = MockWebServer()
    protected val mockResponse = MockResponse()

    @Before
    @Throws(IOException::class)
    fun setup() {
        mockServer.start()
    }

    @After
    @Throws(IOException::class)
    fun tearDown() {
        mockServer.close()
    }

    fun getRetrofit(okHttpClient: OkHttpClient): Retrofit {
        return Retrofit.Builder()
                .baseUrl(mockServer.url("/").toString())
                .client(okHttpClient)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build()
    }

    protected fun getJson(path: String): String? {
        var json: String? = null
        try {
            val inputStream = javaClass.classLoader.getResourceAsStream(path)
            val size = inputStream.available()
            val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()
            json = String(buffer, Charset.forName("UTF-8"))
        } catch (ex: IOException) {
            Log.e("getJson", ex.message)
        }

        return json
    }
}
